workspaces/local-cli/src/shared/spawn-process.ts

```diff
@@ -22,3 +22,26 @@ export async function spawnProcess(
     });
   });
 }
+
+export async function spawnProcessReturnExitCode(
+  command: string,
+  env: any = {}
+): Promise<number> {
+  const taskOptions: SpawnOptions = {
+    env: {
+      ...process.env,
+      ...env,
+    },
+    shell: true,
+    cwd: process.cwd(),
+    stdio: 'inherit',
+  };
+
+  const child = spawn(command, taskOptions);
+
+  return await new Promise((resolve) => {
+    child.on('exit', (code) => {
```

Comment:
```code <number> The exit code if the child exited on its own.
signal <string> The signal by which the child process was terminated.
The 'exit' event is emitted after the child process ends. If the process exited, code is the final exit code of the process, otherwise null. If the process terminated due to receipt of a signal, signal is the string name of the signal, otherwise null. One of the two will always be non-null.```

so I'm not sure when it would exit with a signal in CI but then that's another opportunity for us to erroneously emit exit code 0